modular_skyrat/modules/synths/code/surgery/robot_healing.dm

```diff
@@ -0,0 +1,162 @@
+			if(!..())
+				break
+
+/datum/surgery_step/robot_heal/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/self_message = "You succeed in fixing some of [target]'s damage" //no period, add initial space to "addons"
+	var/other_message = "[user] fixes some of [target]'s damage" //see above
+	var/healed_brute = 0
+
+	if(heals_brute)
+		healed_brute = brute_heal_amount
+		tool.use_tool(target, user, 0, volume=50, amount=1)
+
+	var/healed_burn = 0
+	if(heals_burn)
+		var/obj/item/stack/cable_coil/cable = tool
+		healed_burn = burn_heal_amount
+		if(!cable.amount)
+			return
+		cable.use(1)
+
+	if(missing_health_bonus)
+		if(target.stat != DEAD)
+			healed_brute += round((target.getBruteLoss()/ missing_health_bonus),0.1)
+			healed_burn += round((target.getFireLoss()/ missing_health_bonus),0.1)
+		else //less healing bonus for the dead since they're expected to have lots of damage to begin with (to make TW into defib not TOO simple)
+			healed_brute += round((target.getBruteLoss()/ (missing_health_bonus*5)),0.1)
+			healed_burn += round((target.getFireLoss()/ (missing_health_bonus*5)),0.1)
+
+	if(!get_location_accessible(target, target_zone))
+		healed_brute *= 0.55
+		healed_burn *= 0.55
+		self_message += " as best as you can while they have clothing on"
+		other_message += " as best as they can while [target] has clothing on"
+
+	target.heal_bodypart_damage(healed_brute,healed_burn, 0, BODYTYPE_ROBOTIC)
+	display_results(user, target, span_notice("[self_message]."), "[other_message].", "[other_message].")
+
+	if(istype(surgery, /datum/surgery/robot_healing))
+		var/datum/surgery/robot_healing/the_surgery = surgery
+		the_surgery.surgery_preop_message_sent = TRUE
+
+	return TRUE
+
+/datum/surgery_step/robot_heal/failure(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(
+		user,
+		target,
+		span_warning("You screwed up!"),
+		span_warning("[user] screws up!"),
+		span_notice("[user] fixes some of [target]'s damage."),
+		TRUE,
+	)
+
+	var/brute_damage = 0
+	if(heals_brute)
+		brute_damage = brute_heal_amount * 0.8
+
+	var/burn_damage = 0
+	if(heals_burn)
+		burn_damage = burn_heal_amount * 0.8
+
+	if(missing_health_bonus)
+		brute_damage += round((target.getBruteLoss()/ (missing_health_bonus*2)),0.1)
+		burn_damage += round((target.getFireLoss()/ (missing_health_bonus*2)),0.1)
+
+	target.take_bodypart_damage(brute_damage, burn_damage)
+	return FALSE
+
+/***************************TYPES***************************/
+/datum/surgery/robot_healing/basic
+	name = "Repair robotic limbs (basic)"
+	healing_step_type = /datum/surgery_step/robot_heal/basic
+	desc = "A surgical procedure that provides repairs and maintenance to robotic limbs. Is slightly more efficient when the patient is severely damaged."
+	replaced_by = null
+
+/***************************STEPS***************************/
```

Comment:
Why the separation if there's only one type?

Reply:
Not my code, sadly. Likely left over from the original oldbase code.